From b8f3ea70ce946d05cfa23bf884b4595b7de14298 Mon Sep 17 00:00:00 2001
From: Maxime Chapelet <umxprime@videolabs.io>
Date: Wed, 24 Aug 2022 10:39:30 +0200
Subject: [PATCH 12/14] http : initialize token properly

unititialization caused bad access exceptions when resource was deinit or when its authentication bearer header creation was wrongly attempted
---
 modules/access/http/resource.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/access/http/resource.c b/modules/access/http/resource.c
index 85a0186fc8..d15b1decc4 100644
--- a/modules/access/http/resource.c
+++ b/modules/access/http/resource.c
@@ -208,6 +208,7 @@ int vlc_http_res_init(struct vlc_http_resource *restrict res,
                                                : NULL;
     res->agent = (ua != NULL) ? strdup(ua) : NULL;
     res->referrer = (ref != NULL) ? strdup(ref) : NULL;
+    res->token = NULL;
 
     const char *path = url.psz_path;
     if (path == NULL)
-- 
2.39.3 (Apple Git-145)

